codalab_service.py

```diff
@@ -716,6 +716,12 @@ def build_image(self, image):
         else:
             build_args = ''
 
+        if image == 'frontend' and not self.args.dev:
+            CODALAB_RECAPTCHA_SITE_KEY = os.environ['CODALAB_RECAPTCHA_SITE_KEY']
+            build_args += '--build-arg REACT_APP_CODALAB_RECAPTCHA_SITE_KEY={}'.format(
```

Comment:
Let's do this to simplify instead:
```suggestion
            build_args += '--build-arg REACT_APP_CODALAB_RECAPTCHA_SITE_KEY=$CODALAB_RECAPTCHA_SITE_KEY'
```

Reply:
Actually, even better, I wonder if we could add a parameter to `CodalabArg` called `frontend` -- when this parameter is set, this step will automatically add in all frontend-related variables to build_args. This way, we don't have to duplicate all these variable names.

Reply:
Can we just always pass all the frontend-related variables to build_args? These variables should always be needed right. We can have a list of these variables and loop through those too add them to build_args.

Reply:
Yeah, I was suggesting a similar approach, but rather than maintaining a separate list of frontend-related variables, we can just add a kwarg `frontend=True` that is passed to `CodalabArg` that would let us mark specific arguments as frontend-related. What do you think?

Reply:
I added kwarg and also I switched to use .env.production instead of ARG in dockerfile. This allow us to programmatically dump all the env variables instead of writing them in docker file. `.env.production` will autormatically be picked up when doing `npm build`.